deployment/helm/base/templates/tests/test-namespaces.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: test-namespaces
+  annotations:
+    "helm.sh/hook": test
+
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: kubectl
```

Comment:
How does this `kubectl` command know how to reach the Cluster?  Typically there exists a Kubeconfig file, but I see none mounted in.  Is it inherited somehow?

Reply:
Kubernetes automatically sets up the environment for `kubectl` to use the API server in each container.

The specific service account (`default` in this case) has an associated token that is mounted into the pod at a predefined path, typically `/var/run/secrets/kubernetes.io/serviceaccount/token`. Additionally, Kubernetes sets up certain environment variables inside the container for the `kubectl`
```
> kubectl run env-test --image=busybox --command env
pod/env-test created
> kubectl logs env-test
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=env-test
KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_PORT=443
KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
KUBERNETES_SERVICE_HOST=10.96.0.1
KUBERNETES_SERVICE_PORT=443
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_PORT=tcp://10.96.0.1:443
HOME=/root
```